### PR TITLE
flex_trits_slice: use memcpy for aligned buffers

### PR DESCRIPTION
**Pull request by semenov-vladyslav**
Use `memcpy` in `flex_trits_slice` when buffers are aligned, i.e. `start` is on the boundary of `flex_trit_t`.

# Test Plan:
The changes in the code are similar for TE3, TE4, and TE5; the difference is in handling tails.
`test_flex_trits_slice` in tests seem to cover edge cases and succeed for `trit_encoding` equal 3, 4, and 5.
Simple benchmark works as expected: 10M transaction deserializations work in 7 seconds.

### CODE REVIEW

File: common/trinary/flex_trit.c [hunk 1]
```diff
@@ -25,65 +25,101 @@ size_t flex_trits_slice(flex_trit_t *const to_flex_trits, size_t const to_len, f
...
-    buffer = buffer >> tshift;
-    if (rshift && i < end_index) {
-      buffer |= (flex_trits[i + 1] << rshift);
+  if (0 == (start % NUM_TRITS_PER_FLEX_TRIT)) {
+    // Copy flex trits as bytes
+    memcpy(to_flex_trits, flex_trits + start / NUM_TRITS_PER_FLEX_TRIT, num_trits / NUM_TRITS_PER_FLEX_TRIT);
+    // Handle tail
+    if (0 != num_trits % NUM_TRITS_PER_FLEX_TRIT) {
+      uint8_t buffer = 0;
+      uint8_t shift = (4 - num_trits % NUM_TRITS_PER_FLEX_TRIT) << 1U;
```
Comment (tsvisabo) [inline common/trinary/flex_trit.c]: this code is identical to that of residual calculation (line 88)
can you create a static function that handle that?

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: common/trinary/flex_trit.c (hunk 1 above)
Comment (semenov-vladyslav) [reply common/trinary/flex_trit.c]: I don't know a nice way to do that. Something like:
```
flex_trit_t flex_trits_te4_zero_high_trits(flex_trit_t f, size_t n) {
  // n <= 4
  size_t shift = (4 - n) << 1U;
  f <<= shift;
  f >>= shift;
  return f;
}
```

Comment (semenov-vladyslav) [reply common/trinary/flex_trit.c]: Actually, I had an idea to refactor/rewrite a bit that code. Maybe try to make it more readable.

**Review by tsvisabo — DISMISSED** [review]

**Review by tsvisabo — APPROVED** [review]
